File: api_test/oddsClient.js
require('dotenv').config();
const axios = require('axios');
const fs = require('fs');
const path = require('path');

class OddsAPIClient {
    constructor(apiKey) {
        this.apiKey = apiKey;
        this.baseURL = 'https://api.the-odds-api.com/v4';
        this.requestCount = 0;
    }

    // Get all available sports
    async getSports() {
        return this.makeRequest('/sports');
    }

    // Get odds for a specific sport
    async getOdds(sport, { regions = 'us', markets = 'h2h,spreads', oddsFormat = 'american' } = {}) {
        return this.makeRequest(`/sports/${sport}/odds`, {
            regions,
            markets,
            oddsFormat
        });
    }

    // Make API request
    async makeRequest(endpoint, params = {}) {
        try {
            const response = await axios.get(`${this.baseURL}${endpoint}`, {
                params: {
                    apiKey: this.apiKey,
                    ...params
                }
            });

            this.requestCount++;
            return {
                data: response.data,
                remaining: response.headers['x-requests-remaining'],
                used: response.headers['x-requests-used']
            };
        } catch (error) {
            if (error.response) {
                throw new Error(`API Error: ${error.response.status} - ${JSON.stringify(error.response.data)}`);
            }
            throw error;
        }
    }
}

// Get the next test number
function getNextTestNumber() {
    const baseDir = path.join(process.cwd(), 'test_outputs');
    if (!fs.existsSync(baseDir)) {
        return 1;
    }

    const testDirs = fs.readdirSync(baseDir)
        .filter(name => name.startsWith('test'))
        .map(name => parseInt(name.replace('test', '')))
        .filter(num => !isNaN(num));

    return testDirs.length > 0 ? Math.max(...testDirs) + 1 : 1;
}

// Save response to file
function saveResponse(filename, data, testNumber) {
    const outputDir = path.join(process.cwd(), 'test_outputs', `test${testNumber}`);
    if (!fs.existsSync(outputDir)) {
        fs.mkdirSync(outputDir, { recursive: true });
    }
    const filePath = path.join(outputDir, filename);
    fs.writeFileSync(filePath, JSON.stringify(data, null, 2));
    console.log(`Results saved to: ${filePath}`);
}

// Example usage
async function testOddsAPI() {
    const client = new OddsAPIClient(process.env.ODDS_API_KEY);
    const testNumber = getNextTestNumber();

    try {
        // Get all sports (1 request)
        console.log('\nFetching available sports...');
        const sports = await client.getSports();
        console.log(`Found ${sports.data.length} sports`);
        console.log(`API Requests used: ${client.requestCount}`);
        console.log(`Requests remaining: ${sports.remaining}`);
        saveResponse('1_available_sports.json', sports, testNumber);

        // Get NBA odds (1 request)
        console.log('\nFetching NBA odds...');
        const nbaOdds = await client.getOdds('basketball_nba');
        console.log(`Found ${nbaOdds.data.length} games`);
        console.log(`API Requests used: ${client.requestCount}`);
        console.log(`Requests remaining: ${nbaOdds.remaining}`);
        saveResponse('2_nba_odds.json', nbaOdds, testNumber);

    } catch (error) {
        console.error('Error:', error.message);
        saveResponse('error_log.json', {
            timestamp: new Date().toISOString(),
            error: error.message,
            requestsUsed: client.requestCount
        }, testNumber);
    }
}

// Export the client class and test function
module.exports = {
    OddsAPIClient,
    testOddsAPI
};

// Run test if this file is run directly
if (require.main === module) {
    testOddsAPI();
}

File: wagyu_sports/mcp_server/odds_client.py
#!/usr/bin/env python3
"""
Wagyu Sports Client Module

This module provides a client for interacting with sports betting data APIs.
"""
import requests
from typing import Dict, List, Optional, Any, Union


class OddsClient:
    """
    Client for sports betting data.
    
    This class provides methods for fetching sports betting data including
    available sports and odds for specific sports.
    """
    
    BASE_URL = "https://api.the-odds-api.com/v4"
    
    def __init__(self, api_key: str):
        """
        Initialize the Wagyu Sports client.
        
        Args:
            api_key (str): API key for authentication with The Odds API
        """
        self.api_key = api_key
        self.remaining_requests = None
        self.used_requests = None
    
    def get_sports(self, all_sports: bool = False) -> Dict[str, Any]:
        """
        Get a list of available sports.
        
        Args:
            all_sports (bool, optional): Include out-of-season sports. Defaults to False.
            
        Returns:
            Dict[str, Any]: Response containing available sports data
            
        Raises:
            requests.exceptions.RequestException: If the request fails
        """
        params = {"apiKey": self.api_key}
        if all_sports:
            params["all"] = "true"
            
        return self.make_request("/sports", params)
    
    def get_odds(self, sport: str, options: Optional[Dict[str, Any]] = None) -> Dict[str, Any]:
        """
        Get odds for a specific sport.
        
        Args:
            sport (str): Sport key (e.g., 'basketball_nba')
            options (Dict[str, Any], optional): Additional options for the request. Defaults to None.
                Possible options include:
                - regions: Comma-separated list of regions (e.g., 'us,uk')
                - markets: Comma-separated list of markets (e.g., 'h2h,spreads')
                - oddsFormat: Format for odds ('decimal' or 'american')
                - dateFormat: Format for dates ('unix' or 'iso')
                
        Returns:
            Dict[str, Any]: Response containing odds data
            
        Raises:
            requests.exceptions.RequestException: If the request fails
        """
        endpoint = f"/sports/{sport}/odds"
        params = {"apiKey": self.api_key}
        
        if options:
            params.update(options)
            
        return self.make_request(endpoint, params)
    
    def make_request(self, endpoint: str, params: Optional[Dict[str, Any]] = None) -> Dict[str, Any]:
        """
        Make a request to the sports data API.
        
        Args:
            endpoint (str): API endpoint (e.g., '/sports')
            params (Dict[str, Any], optional): Query parameters. Defaults to None.
            
        Returns:
            Dict[str, Any]: Response data
            
        Raises:
            requests.exceptions.RequestException: If the request fails
        """
        url = f"{self.BASE_URL}{endpoint}"
        response = requests.get(url, params=params)
        
        # Store quota information from headers
        if 'x-requests-remaining' in response.headers:
            self.remaining_requests = response.headers['x-requests-remaining']
        if 'x-requests-used' in response.headers:
            self.used_requests = response.headers['x-requests-used']
        
        # Raise exception for error status codes
        response.raise_for_status()
        
        # Return JSON response
        return {
            "data": response.json(),
            "headers": {
                "x-requests-remaining": self.remaining_requests,
                "x-requests-used": self.used_requests
            }
        }
